sqlalchemy_redshift/dialect.py

```diff
@@ -758,6 +758,13 @@ def _get_all_relation_info(self, connection, **kw):
     @reflection.cache
     def _get_all_column_info(self, connection, **kw):
         all_columns = defaultdict(list)
+
+        generated = (
+            "att.attgenerated as generated"
+            if self.server_version_info >= (12,)
+            else "null as generated"
+        )
```

Comment:
I would be in favour of just using `"null as generated"`. As you alluded to in the OP, we have no reason to believe that Redshift will ever be rebased on a newer version of Postgres, so it doesn't make much sense to check the Postgres version. If generated columns are ever added to Redshift, we could add support at that time.

Reply:
This would also fix the linter errors that are breaking the build.